### ISSUE TITLE
Burp Java Extension Tutorial

### ISSUE THREAD

**Issue by parsiya** [parsiya/parsiya.net#4]
Might become multi-part.

What I have in mind right now:

1. Setting up the development environment.
    * [x] How to setup VScode (e.g., extensions).
    * [x] Gradle file(s) structure and how to set it up. Used `gradle init`.
    * [x] Build with gradle using VSCode tasks. Ctrl+shift+B to run the default task.
    * [x] Add intellicode
    * [x] Add Burp interface (and other projects) to Intellicode via Gradle.
    * [x] Troubleshooting (e.g., run the clean language server task).
2. Creating extensions?
    * [x] This has been beaten to death. There are a total of 123333 tutorials.
    * [ ] Java constructor builder pattern.
        * Maybe use this in a different blog post?
    * [ ] Lessons learned instead.
        * Different blog post.
    * [x] Debugging Java Burp extensions with VS Code (might have to be its own blog post).
        * [x] Apply this to VS Code: https://blog.netspi.com/debugging-burp-extensions/

Add more as I figure it out.

**Comment by parsiya** [comment]
**Debug Burp**
This works.

* `java -agentlib:jdwp=transport=dt_socket,address=localhost:8000,server=y,suspend=n -jar "C:\Program Files\BurpSuiteCommunity\burpsuite_community.jar"`

Adding the command as `-Xagentlib:jdwp=transport=dt_socket,address=localhost:8000,server=y,suspend=n` to the `BurpSuiteCommunity.vmoptions` file does not work because the bundled JRE and running the exe does not have jdwp.

VS Code's `.vscode\launch.json` for debugging:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "java",
            "name": "BurpExtension",
            "request": "attach",
            "hostName": "localhost",
            "port": 8000
        }
    ]
    // How to run Burp
    // java -agentlib:jdwp=transport=dt_socket,address=localhost:8000,server=y,suspend=n -jar "C:\Program Files\BurpSuiteCommunity\burpsuite_community.jar"
}
```

**Comment by parsiya** [comment]
Decent progress over the weekend, should be done tonight.

**Comment by parsiya** [comment]
Fixed in af183f3a8a6c0bbff91f88e5e94478c1d7354350
